spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/firestore/GcpFirestoreAutoConfiguration.java

```diff
@@ -52,10 +57,17 @@
 @EnableConfigurationProperties(GcpFirestoreProperties.class)
 public class GcpFirestoreAutoConfiguration {
 
+	private static final String ROOT_PATH_FORMAT = "projects/%s/databases/(default)/documents";
+
 	private final String projectId;
 
 	private final CredentialsProvider credentialsProvider;
 
+	private final String firestoreRootPath;
```

Comment:
This should probably be configurable through `GcpFirestoreProperties`. WDYT?

Reply:
Hmm one thing I was thinking was to have the autoconfiguration infer the root path with the form `projects/{PROJECT_ID}/databases/(default)/documents` and then extending our template to allow the user to specify a relative path from the root. I don't want the user to have to configure it because it is confusing to figure out the root path format pattern.